### PR TITLE
feat(copy button): Added button to copy example code.

### PR DESCRIPTION
**Pull request by VibhorJaiswal**
Added a button to copy example code without copying it manually. As discussed in #5914 .
Here are the related screenshots.

image
------------------------------------------------------------------------------------------------
image
------------------------------------------------------------------------------------------------
image

### CODE REVIEW

**Review by eirikb — APPROVED** [review]
LGTM

**Review by kyletsang — CHANGES_REQUESTED** [review]

**Comment by VibhorJaiswal** [comment]
1) react-bootstrap `Tooltip` does not scale according to its contents, so when the content inside the `Tooltip` changes it results like this 
image

also using two different `Tooltips ` will decrease code readability significantly. Although I did use the react-bootstrap `Button`.

2) 

> It's better to declare this in a function like `handleBlur` for readability

Done ✅


3) 

> Haven't looked too deeply into react-live, but does the editor expose a way for you to set code as a value so we can use it as a controlled editor?

As far as I know there is a  `onChange` event but when using it, the `react-live` stops showing errors on erroneous code.
Also when I tested, it was working totally fine when using `onBlur` event(also when the code is changed).

File: www/src/components/ReactPlayground.js [hunk 1]
```diff
@@ -219,18 +223,22 @@ function Editor() {
...
   const showMessage = keyboardFocused || (focused && !ignoreTab);
 
+   
+
   return (
     <div className="position-relative">
       <StyledEditor
         onFocus={handleFocus}
-        onBlur={handleBlur}
+        onBlur={(e) => {handleBlur(), handleCodeChange(e)}}
```
[the remark this answers is not in the document]
Comment (VibhorJaiswal) [reply www/src/components/ReactPlayground.js:232]: 1) react-bootstrap `Tooltip` does not scale according to its contents, so when the content inside the `Tooltip` changes it results like this 
image

also using two different `Tooltips ` will decrease code readability significantly. Although I did use the react-bootstrap `Button`.

2) 

> It's better to declare this in a function like `handleBlur` for readability

Done ✅


3) 

> Haven't looked too deeply into react-live, but does the editor expose a way for you to set code as a value so we can use it as a controlled editor?

As far as I know there is a  `onChange` event but when using it, the `react-live` stops showing errors on erroneous code.
Also when I tested, it was working totally fine when using `onBlur` event(also when the code is changed).

**Comment by VibhorJaiswal** [comment]
@kyletsang 
1) react-bootstrap `Tooltip` does not scale according to its contents, so when the content inside the `Tooltip` changes it results like this 
image

also using two different `Tooltips ` will decrease code readability significantly. Although I did use the react-bootstrap `Button`.

2) 

> It's better to declare this in a function like `handleBlur` for readability

Done ✅


3) 

> Haven't looked too deeply into react-live, but does the editor expose a way for you to set code as a value so we can use it as a controlled editor?

As far as I know there is a  `onChange` event but when using it, the `react-live` stops showing errors on erroneous code.
Also when I tested, it was working totally fine when using `onBlur` event(also when the code is changed).

**Comment by kyletsang** [comment]
> react-bootstrap `Tooltip` does not scale according to its contents, so when the content inside the `Tooltip` changes it results like this
>

Might be able to use this for the dynamic content:
https://react-bootstrap.github.io/components/overlays/#overlays-dynamic-updates

**Comment by VibhorJaiswal** [comment]
@kyletsang 
1) > Can we use existing react-bootstrap components here (Button and Tooltip)? That way, you wouldn't have to define new styles for the button/tooltip. ✅

2) > It's better to declare this in a function like `handleBlur`for readability. ✅

File: www/src/components/ReactPlayground.js [hunk 2]
```diff
@@ -256,11 +267,39 @@ const propTypes = {
...
+  const handleCodeChange = (e) => { setcodeToCopy(e.target.value) }
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(codeToCopy) //copies code to clipboard
+      .then(setCopy('Copied!'))
+
+  }
+
+
+  const UpdatingPopover = React.forwardRef(
```
Comment (kyletsang) [inline www/src/components/ReactPlayground.js:287]: Components shouldn't be nested within another component.  This should be declared outside.

File: www/src/components/ReactPlayground.js [hunk 3]
```diff
@@ -269,8 +308,28 @@ function Playground({ codeText, exampleClassName, showCode = true }) {
...
+
+          <>
+            <div onMouseOut={resetCopyStatus}>
+              <OverlayTrigger
+                trigger={['hover', 'focus']}
+                overlay={
+                  <UpdatingPopover id="popover-contained">{copyStatus}</UpdatingPopover>
+                }
+              >
+                <Button onClick={handleCopy} className={styles.styledCopyButton} variant='dark' value={codeText} >
```
Comment (kyletsang) [inline www/src/components/ReactPlayground.js:325]: Is `value` needed here?

**Review by kyletsang — COMMENTED** [review]
Looking good.  Couple more comments.  Can you also run prettier on the files you've changed?  I'm noticing there's some style inconsistencies.

**Comment by VibhorJaiswal** [comment]
@kyletsang 
> Can you also run prettier on the files you've changed? I'm noticing there's some style inconsistencies.

I have prettified according to my prettier, if this is not what you want then can you please help me in this case.

**Comment by SaltyBuckets** [comment]
> Can you also run prettier on the files you've changed? 

@kyletsang  tbh you should add a bot to do that to decrease the load on new contributers

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
I think this is a good first pass.  @jquense, want to review?

**Comment by kyletsang** [comment]
Sorry for delay on this.  Thanks for PR!

**Comment by VibhorJaiswal** [comment]
@kyletsang no problem - thanks

**Pull request merged**